rust/datafusion/src/logicalplan.rs

```diff
@@ -419,13 +410,50 @@ impl Expr {
         }
     }
 
+    /// Determine if this expression can produce null values
+    pub fn nullable(&self, input_schema: &Schema) -> Result<bool> {
+        match self {
+            Expr::Alias(expr, _) => expr.nullable(input_schema),
+            Expr::Column(name) => Ok(input_schema.field_with_name(name)?.is_nullable()),
+            Expr::Literal(value) => match value {
+                ScalarValue::Null => Ok(true),
+                _ => Ok(false),
+            },
+            Expr::Cast { expr, .. } => expr.nullable(input_schema),
+            Expr::ScalarFunction { .. } => Ok(true),
+            Expr::AggregateFunction { .. } => Ok(true),
```

Comment:
I think the rules here are subtle:
If the aggregate is MIN/MAX/SUM then the nullability is the nullability of the input
If the aggregate is COUNT then the nullability is false (COUNT never can produce NULL)

However, `true` is the safe default, so I think this is ok